### PR TITLE
Quote args containing spaces in ArgsStringOrList

### PR DESCRIPTION
**Pull request by tartarughina**
Address #194

When ArgsStringOrList::List elements contain spaces, they are now wrapped in double quotes before joining. This prevents the debugger from splitting paths with spaces into multiple arguments.

Add tests covering list with spaces, list without spaces, single element with spaces, empty list, and single string passthrough.

Tested with debug entry

```
{
    "label": "Args as list - path with spaces",
    "adapter": "Java",
    "request": "launch",
    "mainClass": "ArgsTest",
    "args": ["C:\\path with spaces\\some file.txt", "arg2"],
    "build": {
      "command": "javac ArgsTest.java",
      "shell": {
        "with_arguments": {
          "program": "/bin/sh",
          "args": ["-c"]
        }
      }
    }
  }
```

and verified that the provided args with spaces are treated as a single string

image

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -429,7 +429,74 @@ impl Serialize for ArgsStringOrList {
...
         match self {
             ArgsStringOrList::String(s) => serializer.serialize_str(s),
-            ArgsStringOrList::List(l) => serializer.serialize_str(&l.join(" ")),
+            ArgsStringOrList::List(l) => {
+                let quoted: Vec<String> = l
+                    .iter()
+                    .map(|s| {
+                        if s.contains(' ') {
+                            format!("\"{}\"", s)
+                        } else {
```
Comment (playdohface) [inline src/util.rs]: My gut says why not just wrap it in quotes unconditionally, but I remember there was a gotcha there. And, I know this would be madness, but can paths technically contain tabs? Like, is the `contains(' ')` check exhaustive here?
